zork.js

```diff
@@ -0,0 +1,329 @@
+                }
+            }
+
+/*             if(keypad.actions.includes(prompt)){
+                console.log("Enter the next building.")
+                currentLocation = foyer182Main
+            } */
+        }
+    }
+
+    if(currentLocation === foyer182Main){
+        let current = await ask(foyer182Main.description)
+
+        if(current.startsWith('take')){
+            takeItem(newPlayer,current)
+            //newPlayer.inventory.push(current.substring(4))
+            console.log("Plyaer inv = " + newPlayer.inventory)
+        }
+        if(current === 'read'){
+            console.log("You read the paper.... ")
+            current = await ask(foyer182Main.description)
+        }
+        if(current === 'go upstairs'){
+            currentLocation = currentLocation.nextLocation
+        }
+    }
+
+    if(currentLocation === BCA){
+        let cur = await ask(currentLocation.description)
+        //console.log("CURRENT LOCATION === " + currentLocation.name)
+        if(currentLocation.items.includes(cur)){
+            //TODO manually set next location here... for some reason
+            currentLocation = BCAclassroom
+        }
+    }
+
+    if(currentLocation === BCAclassroom){
+        let cur = await ask(currentLocation.description)
+        if(currentLocation.items[0].actions.includes(cur)){
+            //pickup computer 
+            takeItem(newPlayer,currentLocation.items[0])
+            console.log("New player iteems = " + newPlayer.inventory)
+        }
+        if(cur === 'leave class' || cur === 'leave'){
+            currentLocation = currentLocation.nextLocation
+        }
+    }
+
+    if(currentLocation === churchStreet){
+        let cur = await ask(currentLocation.description)
+        if(currentLocation.items.includes(cur)){
+            console.log("Great idea, you walk tot the lakefront")
+            currentLocation = lakefront
+        }
+    }
+
+    if(currentLocation === lakefront){
+        let cur = await ask(currentLocation.description)
+        if(currentLocation.items.includes(cur)){
+            currentLocation = splash
+        }
+    }
+
+    if(currentLocation === splash){
+        let cur = await ask(currentLocation.description)
+        if(cur === 'take pint glass'){  
+            console.log("Bob wants a glass also, so you give him one.")
+            giveItem(newPlayer,bob,cur.substring(4))
+            takeItem(newPlayer,cur.substring(4))
+        }
+        if(cur === 'go home'){
+            currentLocation = home
+        }
+        
+    }
+
+    if(currentLocation === home){
+        let cur = await ask(home.description)
+        if(cur === 'go to work'){
+            currentLocation = truck
+        }
+        if(cur === 'get in truck'){
+            currentLocation = truck
+        }
+    }
+
+    if(currentLocation === truck){
+        let locationsPossible = [];
+        //TODO should map these objects to "NAME": "locationObject"
```

Comment:
If you want to map the `locationObject` value to a string version of the name you may want to consider using an object instead of an array since objects are collections of key:value pairs where the keys are strings.